sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java

```diff
@@ -502,6 +521,77 @@ public HttpPipeline getHttpPipeline() {
+     * Returns a reactive Publisher emitting all the files/directories in this filesystem that have been recently soft
+     * deleted lazily as needed. For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/list-blobs">Azure Docs</a>.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileSystemAsyncClient.listDeletedPaths}
+     *
+     * @return A reactive response emitting the list of files/directories.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<PathDeletedItem> listDeletedPaths() {
+        try {
+            return this.listDeletedPaths(new ListDeletedPathsOptions());
+        } catch (RuntimeException ex) {
+            return pagedFluxError(logger, ex);
+        }
+    }
+
+    /**
+     * Returns a reactive Publisher emitting all the files/directories in this account lazily as needed. For more
+     * information, see the <a href="https://docs.microsoft.com/rest/api/storageservices/datalakestoragegen2/filesystem/list">Azure Docs</a>.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileSystemAsyncClient.listDeletedPaths#ListDeletedPathsOptions}
+     *
+     * @param options A {@link ListDeletedPathsOptions} which specifies what data should be returned by the service.
+     * @return A reactive response emitting the list of files/directories.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<PathDeletedItem> listDeletedPaths(ListDeletedPathsOptions options) {
+        return listDeletedPathsWithOptionalTimeout(options, null);
+    }
+
+    PagedFlux<PathDeletedItem> listDeletedPathsWithOptionalTimeout(ListDeletedPathsOptions options,
+        Duration timeout) {
+        Function<String, Mono<PagedResponse<PathDeletedItem>>> func =
+            marker -> listDeletedPathsSegment(marker, options, timeout)
+                .map(response -> {
+                    List<PathDeletedItem> value = response.getValue().getSegment() == null
+                        ? Collections.emptyList()
+                        : Stream.concat(
+                        response.getValue().getSegment().getBlobItems().stream().map(Transforms::toPathDeletedItem),
+                        response.getValue().getSegment().getBlobPrefixes().stream()
+                            .map(Transforms::toPathDeletedItem)
+                    ).collect(Collectors.toList());
+
+                    return new PagedResponseBase<>(
+                        response.getRequest(),
+                        response.getStatusCode(),
+                        response.getHeaders(),
+                        value,
+                        response.getValue().getNextMarker(),
+                        response.getDeserializedHeaders());
+                });
+
+        return new PagedFlux<>(() -> func.apply(null), func);
```

Comment:
Does listing deleted paths support page size?

Reply:
ooh yeah good catch. I need to port over Rick's recent changes to support pageSize here

Reply:
Added support